native/rd-landlock/src/sys.rs

```
@@ -11,6 +11,344 @@ use std::process::Command;
+    rules: &[SocketFilterRule],
+) -> Result<Vec<libc::sock_filter>, String> {
+    let arch = arch_from_rules(rules)?;
+    let socket_families = allowed_socket_families(rules, syscall_socket());
+    let socket_errno = default_socket_family_errno(rules, syscall_socket())?;
+    let socketpair_families = allowed_socket_families(rules, syscall_socketpair());
+    let socketpair_errno = default_socket_family_errno(rules, syscall_socketpair())?;
+    let denied_syscalls = denied_syscalls(rules);
+    #[cfg(target_arch = "x86_64")]
+    let denied_syscall_masks = denied_syscall_number_masks(rules);
+
+    let mut program = Vec::new();
```

Clippy (1.96.0)

[warning] calls to `push` immediately after creation

(warning)